implemented  skeletons  loading page, for all navigation

Saurabhsing21 opened the pull request:
/claim #533
Fixes #533 

# Implementing Skeleton Loading for Navigation and Images

Skeleton loading is a technique used to display a placeholder while content is being loaded, improving user experience by reducing perceived load times. Below is how to implement skeleton loading for **Navigation** and **Images**.

### Before I:
![image alt](https://github.com/Saurabhsing21/test-learn/blob/main/loading.png?raw=true)

### After:
![image alt](https://github.com/Saurabhsing21/hodlinfo-task/blob/main/Previw.mov)

Saurabhsing21 commented:
/claim #533

Saurabhsing21 commented:
> Not the right fix, the loading will still appear.
> 
> The reason is the loading appears is because index.html loads before other assets

Thanks, I'm just  done with changes you require. i will submit it soon

Saurabhsing21 commented:
@seveibar  i fixed  with  complete skelton loading page in index.html. please review this, let me know if any other changes are require

imrishabh18 commented on index.html:
```diff
@@ -15,9 +15,202 @@
     <script type="module" src="/src/main.tsx"></script>
+
+    <!-- Script to Hide Skeleton -->
+    <script>
+      document.addEventListener("DOMContentLoaded", () => {
+        const skeletonLoader = document.getElementById("skeleton-loader");
+        if (skeletonLoader) {
+          console.log("Skeleton Loader is displayed.");
+          setTimeout(() => {
+            skeletonLoader.style.display = "none";
+          }, 300); // Hide skeleton after React initializes
+        }
+      });
+    </script>
```
@Saurabhsing21 I don't we need the cleanup and hiding of the skeleton at both the places, I see you have added this in the `useEffect()` hook as well.

Saurabhsing21 replied:
@imrishabh18  just to clarify, do you mean that I should remove the script tag and keep the skeleton hiding logic only in the useEffect() hook?

imrishabh18 replied:
Yeah, give it a try. I think that should work

imrishabh18 commented:
Other than that, this LGTM! Make sure you are fixing the format issues

Saurabhsing21 commented:
> Other than that, this LGTM! Make sure you are fixing the format issues

ok i will fix this

Saurabhsing21 commented:
@imrishabh18 please review this,

imrishabh18 commented:
LGTM!

cc: @seveibar

seveibar reviewed (approved):
LGTM, let's see @ShiboSoftwareDev weigh in!!!!

ShiboSoftwareDev reviewed (changes requested):
I didn't give time to the ai page to preload and this is what I got:

https://github.com/user-attachments/assets/507af085-5744-48c4-a972-9f002739e2d6

Saurabhsing21 commented:
> I didn't give time to the ai page to preload and this is what I got:
> 
>  Screen.Recording.2025-01-16.193042.mp4

  sorry for my mistake. I will  change as per requirement.

Saurabhsing21 commented:
@ShiboSoftwareDev please review  this

seveibar commented:
@Saurabhsing21 I'll defer to @ShiboSoftwareDev  on this one!

ShiboSoftwareDev commented:
@Saurabhsing21 
I'm sorry I didn't say this earlier but a skeleton is not what I expected from this issue, but I said I'd give it a chance maybe it'd look good.
A skeleton page gives the illusion of receiving the page as soon as you request it for faster user feedback.
Since all our pages have a different layout this would be hard to maintain, and when I saw the current implementation the skeleton got drawn in my memory and when the pages loaded it looked different which wasn't a good experience.

The loading screen should have a simple loading animation that is consistent with the project, just a pleasant animation to keep the user intrigued.

Saurabhsing21 commented:
> @Saurabhsing21 I'm sorry I didn't say this earlier but a skeleton is not what I expected from this issue, but I said I'd give it a chance maybe it'd look good. A skeleton page gives the illusion of receiving the page as soon as you request it for faster user feedback. Since all our pages have a different layout this would be hard to maintain, and when I saw the current implementation the skeleton got drawn in my memory and when the pages loaded it looked different which wasn't a good experience.
> 
> The loading screen should have a simple loading animation that is consistent with the project, just a pleasant animation to keep the user intrigued.

@ShiboSoftwareDev  i have sent an email regarding this please check

ShiboSoftwareDev commented:
@Saurabhsing21 I'm not great at design, I just want something consistent with the project and pleasant to watch

Saurabhsing21 commented:
@ShiboSoftwareDev, can you review this? I have added a progress bar as LinkedIn has  with animation  over text ; this was suggested by Steve. 

https://github.com/user-attachments/assets/38450b3d-c3a3-4cbf-ab32-eb75072df1bd

Saurabhsing21 commented:
@seve  review this

Saurabhsing21 commented:
Uploading Screen Recording 2025-01-20 at 11.07.21 PM.mov…

Saurabhsing21 commented:
https://github.com/user-attachments/assets/01587fed-1cdf-47c5-9038-055a196baf0d

seveibar commented on src/App.tsx:
```diff
@@ -82,10 +80,27 @@ class ErrorBoundary extends React.Component<
+      )
+    }
+    return null
+  }
```
why is this here

seveibar reviewed (changes requested):
@ShiboSoftwareDev correct if i'm wrong, but there should be no changes to App.tsx, this loader should live exclusively in index.html?

ShiboSoftwareDev commented:
@seveibar I don't know why you think that, it works and the suspense component is made for this kind of thing

seveibar commented:
@Saurabhsing21 @ShiboSoftwareDev 

ok so here's my take: The best loading page will do this inside `index.html`. Why? Because the page makes requests in this order:

1. index.html
2. App.js (chunk)
3. Page1.js, Page2.js, Page3.js (chunks for pages)

So what shows in the middle? If you use suspense, this happens:

- user sees: blank
- index.html loaded
- user sees: blank
- App.js (chunk)
- user sees: Loading Page
- Other pages load
- user sees: Correct Page


If you put the loader in index.html, this happens:

- user sees: blank
- index.html loaded
- **user sees: Loading Page**
- App.js (chunk)
- user sees: Loading Page
- Other pages load
- user sees: Correct Page

Saurabhsing21 commented:
@seveibar  i have solved this in index.html completely, please check once

ShiboSoftwareDev reviewed (changes requested):
@Saurabhsing21 and now it doesn't work, can't you see the preview? I think the problem is that it's using the suspense component even though it's null so it's displaying an empty page, maybe removing the suspense component would fix it?

Saurabhsing21 commented:
> @Saurabhsing21 and now it doesn't work, can't you see the preview? I think the problem is that it's using the suspense component even though it's null so it's displaying an empty page, maybe removing the suspense component would fix it?

i  checked  it is working here is  preview 

Uploading Screen Recording 2025-01-21 at 4.37.25 PM.mov…

Saurabhsing21 commented:
> @Saurabhsing21 and now it doesn't work, can't you see the preview? I think the problem is that it's using the suspense component even though it's null so it's displaying an empty page, maybe removing the suspense component would fix it?

i checked it is working here is preview

https://github.com/user-attachments/assets/1cf1ba9c-aa5a-45f1-954d-aca90878bdd9

ShiboSoftwareDev commented:
the preview is this: 
![image](https://github.com/user-attachments/assets/8d838bfc-ba85-406d-8ad7-4e635da9b1c1)

ShiboSoftwareDev commented:
vite is different in development and production

Saurabhsing21 commented:
@seveibar @ShiboSoftwareDev i have implemented feature entirly in index.html it is working in development but not in deployment. I tried to remove suspense components as Shibo suggested, but it is giving me a playwright issue. Can you suggest how I can handle this?

ShiboSoftwareDev commented:
@Saurabhsing21 it still doesn't work so don't worry about playwright atm

Saurabhsing21 commented:
> @Saurabhsing21 it still doesn't work so don't worry about playwright atm

Then how to solve this development vs production problem 
@seveibar  ?

Saurabhsing21 commented:
> > @Saurabhsing21 it still doesn't work so don't worry about playwright atm
> 
> Then how to solve this development vs production problem @seveibar ?



> > @Saurabhsing21 it still doesn't work so don't worry about playwright atm
> 
> Then how to solve this development vs production problem. @seveibar ?

Saurabhsing21 commented:
@ShiboSoftwareDev  review this i have fixed  deployment problem

ShiboSoftwareDev commented:
@Saurabhsing21 when I go to the ai page it appears for a little and then disappears

Saurabhsing21 commented:
> @Saurabhsing21 when I go to the ai page it appears for a little and then disappears
@ShiboSoftwareDev 
Sorry, but I did not understand at which appearance you are talking about. Can you give any image or video, especially what is happening and what I need to change?

ShiboSoftwareDev commented:
@Saurabhsing21 when I click the AI tab I see the loading screen and then it disappears leaving an empty page before the AI page is loaded

Saurabhsing21 commented:
@ShiboSoftwareDev   i have removed blank  page problem, please check once another thing i want you to chack loading animation for same page navigation, just like LinkedIn and other webpages the use skeleton loading animation for same  page navigation  so that also implemented same loading animation for  same  page navigation chec. let me know if there any changes are needed

ShiboSoftwareDev commented:
@Saurabhsing21 


https://github.com/user-attachments/assets/71d09553-ede0-4704-ae02-62cd786eff01

Saurabhsing21 commented:
> This PR has been automatically marked as stale because it has had no recent activity. It will be closed if no further activity occurs.

i  am working on it
